massa-hash/src/hash.rs

```diff
@@ -102,14 +111,10 @@ impl Hash {
     /// # use massa_hash::hash::Hash;
     /// let hash = Hash::compute_from(&"hello world".as_bytes());
     /// let serialized = hash.into_bytes();
-    /// let deserialized: Hash = Hash::from_bytes(&serialized).unwrap();
+    /// let deserialized: Hash = Hash::from_bytes(&serialized);
     /// ```
-    pub fn from_bytes(data: &[u8; HASH_SIZE_BYTES]) -> Result<Hash, MassaHashError> {
-        use bitcoin_hashes::Hash;
-        Ok(Hash(
-            bitcoin_hashes::sha256::Hash::from_slice(&data[..])
-                .map_err(|err| MassaHashError::ParsingError(format!("{}", err)))?,
-        ))
+    pub fn from_bytes(data: &[u8; HASH_SIZE_BYTES]) -> Hash {
```

Comment:
What happens if we receive some random bytes that we try to parse as a hash ?

Reply:
The function will parse a `Hash` no matter what, if you wonder why the previous implementation returned a `Result<T, E>` it was because `bitcoin_hashes::sha256::Hash::from_slice` does not take a fixed-sized array and therefore has to check for the length (see [here](https://docs.rs/bitcoin_hashes/latest/src/bitcoin_hashes/sha256.rs.html#133-141)). Since Blake3 conversion takes a fixed sized array as input it is not necessary to have that kind of check or error.